tests/unit/misc/test_lineparser.py

```diff
@@ -53,24 +54,89 @@ def test_prepare_save_missing(self, mocker, lineparser):
         lineparser._prepare_save()
         os_mock.makedirs.assert_called_with(self.CONFDIR, 0o755)
 
+    def test_prepare_save_no_config(self, mocker):
+        """Test if _prepare_save doesn't create a None config dir."""
+        os_mock = mocker.patch('qutebrowser.misc.lineparser.os')
+        os_mock.path.exists.return_value = True
+
+        lineparser = lineparsermod.BaseLineParser(None, self.FILENAME)
+        assert not lineparser._prepare_save()
+        assert not os_mock.makedirs.called
+
+    def test_double_open(self, mocker, lineparser):
+        """Test if _open refuses reentry."""
+        mocker.patch('builtins.open', mock.mock_open())
+
+        with lineparser._open('r'):
+            with pytest.raises(IOError):
```

Comment:
Maybe do `with pytest.raises(IOError) as excinfo:` and `assert str(excinfo.value) == "..."` at the end to check the exception message?